### DISCUSSION TITLE
Why open api skills return a RestApiOperationResponse instead of raw content ?

### DISCUSSION THREAD

**Discussion by Poltuu** [microsoft/semantic-kernel#1871] [General]
The result of an api call from openapi skill returns a `RestApiOperationResponse`. I was wondering why not return directly the HTTP content ?

My main concerne is that the actual ouput is actually a JSON string, so if the content from the api call is also an JSON (very common case), it end up being escaped, which leads to very bloated output, and being more difficult to read in a following step of a planner for instance.

**Comment by craigomatic** [comment]
@SergeyMenshykh any comments you would add here?

**Reply by SergeyMenshykh** [reply]
Using RestApiOperationResponse instead of HTTP content provides a flexible model that is easily extensible to support new requirements for additional metadata or data on top of the HTTP content itself. Moreover, it currently has a 'contentType' field that describes the returned content and can be used in 'if' statements if a decision needs to be made based on the content type.

**Reply by Poltuu** [reply]
I understand and agree with your statement. However in the current form, the Json containing another escaped Json payload is proving to be simply unusable for another semantic function, as the simple size and unreadability of it means it implies an extra bit of preprocessing is needed, which the planner today cannot think about ahead of time, if that makes sense.
